File: Programers/Level2/weekly/전력망둘로나누기.py
from collections import deque


def chkWires(wires, idx):
    q = deque()
    arrList = []
    cnt = 0
    for arr in wires :
        arrList.append(arr)
        
    q.append(arrList[idx][1])
    arrList.pop(idx)
    
    while q:
        
        wireNum = q.popleft()
        for i in range(0, len(arrList)):
            
            flag = False
            temp = arrList[i]
            
            if wireNum == temp[0] :
                flag = True
                q.append(temp[1])
                arrList.remove(temp)
                # cnt += 1
                
            elif wireNum == temp[1] :
                flag = True
                q.append(temp[0])
                arrList.remove(temp)
                # cnt += 1
            
            if flag:
                i -= 1
        cnt += 1
    return cnt
        
    return answer

def solution(n, wires):
    answer = 1001
    
    for i in range(len(wires)):
        leftWire = chkWires(wires, i)
        rightWire = n - leftWire
        
        temp = abs(leftWire - rightWire)
        answer = min(answer, temp)
        
    return answer

File: Javascript/Programmers/전력망둘로나누기.js
function solution(n, wires) {
    let minValue = 10001;

    for (let i = 0; i < wires.length; i++) {
        const leftWire = chkWire(wires, i);
        const rightWire = n - leftWire;
        const temp = Math.abs(leftWire - rightWire);
        minValue = Math.min(minValue, temp);
    }

    return minValue;
}

function chkWire(wires, idx) {
    let cnt = 0;
    const w_list = [...wires];

    q = [];
    q.push(w_list[idx][1]);
    w_list.splice(idx, 1);

    while (q.length > 0) {
        const k = q.shift();

        for (let i = 0; i < w_list.length; i++) {
            let flag = false;
            let temp = w_list[i];

            if (temp[0] === k) {
                q.push(temp[1]);
                // w_list.filter(v => v !== temp);
                for (let i = 0; i < w_list.length; i++) {
                    if (w_list[i] === temp) {
                        w_list.splice(i, 1);
                        i--;
                    }
                }
                flag = true;
            } else if (temp[1] === k) {
                q.push(temp[0]);
                // w_list.filter(v => v !== temp);
                for (let i = 0; i < w_list.length; i++) {
                    if (w_list[i] === temp) {
                        w_list.splice(i, 1);
                        i--;
                    }
                }
                flag = true;
            }

            if (flag) {
                i -= 1;
            }
        }
        cnt += 1;
    }
    return cnt;
}
